harvester/file.py

```diff
@@ -61,13 +61,30 @@ def file_from_element(cls, file_element, binding_dc_identifier):
         location = file_element.xpath("./*/@*[local-name()='href']")[0]
         parent = file_element.getparent()
 
-        if (
-            parent.attrib["USE"] in ["alto", "Text"]
-            and parent.attrib["ID"] == "ALTOGRP"
-        ):
+        filegrp_use = parent.attrib["USE"]
+        filegrp_id = parent.attrib["ID"]
+
+        if filegrp_use in ["alto", "Text"] and filegrp_id == "ALTOGRP":
             file_cls = ALTOFile
+        elif filegrp_use in ["Images", "reference"] and filegrp_id in [
```

Comment:
This sort of thing is maybe more reliable with enums, as in `from enum import Enum`. And then you can't have unhandled types. But not necessary. Come to think of it, the same is true of how I made the paths in the `download_images` dag be a dict, they should really be enums too to protect against typos.

Reply:
What would you make the enum names and values? If we based our decision on one attribute only, we could do something like
```
class FileGrpID(Enum):
    ALTOGRP = ALTOFile
    IMGGRP = AccessImageFile
    ACIMGGRP = AccessImageFile
    ...

file_cls = FileGrpIP.filegrp_id.value
```
but if we make decisions based on both the ID and USE attributes, things get hairy as tuples or such aren't a legal enum members.

Or do you mean that we would have enums along the lines of
```
class FileGrpIDs(Enum):
    ALTO = ["ALTOGRP"]
    ACCESS_IMAGE = ["IMGGRP", "ACIMGGRP"]
    SKIPPED_FILE = ["TARGETIMGGRP", "WHATEVER", "SOMETHING"]
    ...

class FileGrpUses(Enum):
    ALTO = [”alto", "Text"]
    ...
```
and then
```
if filegrp_use in FileGrpUses.ACCESS_IMAGE.values and filegrp_id in FileGrpIDs.ACCESS_IMAGE.values:
    file_cls = AccessImageFile
elif ...
```

Reply:
Ah, never mind, for some reason when I was reading this I thought this was our internal code, not XML parsing code, this makes sense. I retract my earlier comment and substitute it for this: do we really want to raise an exception for future unknown file groups, rather than skip them?

Reply:
That's a great question. And I'm not sure of the best answer. On the other hand, raising the exception means that we will have a red pipeline if (when?) NLF introduces a new file group (downloadable or not) and we need to manually categorize it. But on the other hand, if they do that and we don't raise an exception, the pipeline can silently produce an incomplete data set with some files missing for some bindings, and currently we have no other mechanisms in place that would alert us (before some user complains).

Reply:
We decided that since this will still result in retrying and ultimately moving on, this is fine. Although now we *won't* be very strongly alerted to the situation unless we pay attention to logs.